outscale/resource_outscale_access_key.go

```diff
@@ -170,3 +205,17 @@ func updateAccessKey(conn *oscgo.APIClient, id, state string) error {
 
 	return nil
 }
+
+func checkDateFromat(dateFormat string) error {
+	var err error
+	currentDate := time.Now()
+	settingDate := time.Now()
+
+	if settingDate, err = datetime.Parse(dateFormat, time.UTC); err != nil {
+		return fmt.Errorf("Expiration Date should be 'ISO 8601' format ('2017-06-14' or '2017-06-14T00:00:00Z, ...) %s", err)
+	}
+	if currentDate.After(settingDate) || settingDate == currentDate {
```

Comment:
currentDate.Equal(settingDate) maybe ? According to the documentation, == does not check Location (cf https://pkg.go.dev/time#Time.Equal)